### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by mykolahuryn**
[DEMO LINK](https://mykolahuryn.github.io/js_employees_table_DOM/)

### CODE REVIEW

File: src/scripts/main.js [hunk 1]
```diff
@@ -1,3 +1,300 @@
 'use strict';
 
-// write code here
+const table = document.querySelector('table');
+const tbody = document.querySelector('tbody');
+const sortTableASC = function(index, type) {
```
Comment (Anastasiia-Svintsova) [inline src/scripts/main.js:5]: Reduce code duplication: The sortTableASC and sortTableDESC functions are almost the same. So just combine it to a one sorting function

File: src/scripts/main.js [hunk 2]
```diff
@@ -1,3 +1,300 @@
...
+        } else if (rowADate > rowBDate) {
+          return 1;
+        }
+
+        return 0;
+      case 'Age':
+        return rowADate - rowBDate;
+      case 'Salary':
+        const dateA = rowADate.split('$').join('').split(',').join('');
+        const dateB = rowBDate.split('$').join('').split(',').join('');
```
Comment (Anastasiia-Svintsova) [inline src/scripts/main.js:25]: You can create a separate method for convert this value

**Review by Anastasiia-Svintsova — CHANGES_REQUESTED** [review]
- Add min values for these inputs
image

 - Can add an employee with invalid name and position  fields 

https://user-images.githubusercontent.com/99403198/229286306-7770ca2c-1b44-4d7a-b4c6-f8594d57fa27.mov

- Also I can add invalid values during editing 😢 

https://user-images.githubusercontent.com/99403198/229286337-997f8a5b-533d-48c7-9031-28edddc98c45.mov

**Comment by mykolahuryn** [comment]
[DEMO LINK](https://mykolahuryn.github.io/js_employees_table_DOM/)

File: src/scripts/main.js [hunk 3]
```diff
@@ -1,3 +1,289 @@
...
+    const rowBDate = rowB.cells[index].innerHTML;
+
+    switch (type) {
+      case 'Name':
+      case 'Position':
+      case 'Office':
+        if (rowADate < rowBDate) {
+          return isAscending ? -1 : 1;
+        } else if (rowADate > rowBDate) {
+          return isAscending ? 1 : -1;
```
Comment (polosanya) [inline src/scripts/main.js:20]: It is better to compare strings using localeCompare method

File: src/scripts/main.js [hunk 4]
```diff
@@ -1,3 +1,289 @@
...
+        return 0;
+      case 'Age':
+        return isAscending ? rowADate - rowBDate : rowBDate - rowADate;
+      case 'Salary':
+        const dateA = rowADate.replace(/[$,]/g, '');
+        const dateB = rowBDate.replace(/[$,]/g, '');
+
+        return isAscending ? dateA - dateB : dateB - dateA;
+    }
+  };
```
Comment (polosanya) [inline src/scripts/main.js:32]: Don't forget about default case for error handling
Fix all cases

**Review by polosanya — CHANGES_REQUESTED** [review]
It would be great to specify the messages on editing, not just "success message" and "error message". Also, make sure that notifications overlay the layout (set the proper z-index)
table

**Comment by mykolahuryn** [comment]
[DEMO LINK](https://mykolahuryn.github.io/js_employees_table_DOM/)

File: src/scripts/main.js [hunk 5]
```diff
@@ -1,3 +1,301 @@
...
+      case 'Office':
+        return isAscending ? rowADate.localeCompare(rowBDate)
+        : rowBDate.localeCompare(rowADate);
+      case 'Age':
+        return isAscending ? rowADate - rowBDate : rowBDate - rowADate;
+      case 'Salary':
+        const dateA = rowADate.replace(/[$,]/g, '');
+        const dateB = rowBDate.replace(/[$,]/g, '');
+
+        return isAscending ? dateA - dateB : dateB - dateA;
```
Comment (agniya-i) [inline src/scripts/main.js:25]: Not fixed, please add the default case

Screenshot 2023-04-17 at 18 30 02

```suggestion
        return isAscending ? dateA - dateB : dateB - dateA;
        
        default: 
        ...
        
```

**Review by agniya-i — CHANGES_REQUESTED** [review]

File: src/scripts/main.js (hunk 5 above)
Comment (mykolahuryn) [reply src/scripts/main.js:25]: @agniya-i  I'm sorry, I don't fully understand what this means (

**Review by agniya-i — APPROVED** [review]

**Pull request closed without merging**